sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SemanticModelHelper.cs

```diff
@@ -27,9 +27,14 @@ public static class SemanticModelHelper
         public static SemanticModel GetSyntaxTreeSemanticModel(this SemanticModel model, SyntaxNode node)
         {
             // See https://github.com/dotnet/roslyn/issues/18730
-            return model.SyntaxTree == node.SyntaxTree
-                ? model
-                : model.Compilation.GetSemanticModel(node.SyntaxTree);
+            if (model.SyntaxTree == node.SyntaxTree)
+            {
+                return model;
+            }
+
+            return model.Compilation.ContainsSyntaxTree(node.SyntaxTree)
+                ? model.Compilation.GetSemanticModel(node.SyntaxTree)
+                : null;
```

Comment:
It seems to be possible not to have a semantic model for a given syntax tree.

Reply:
Since this method can now return _null_, should it be renamed _FindXXX_?

It might also be worth adding a comment explaining why this might happen e.g. when running in the IDE it's possible to retrieve nodes that belong to other compilations.

Reply:
What about this review comment?

Reply:
Sorry I missed it. I am not a big fan of `Find` except on collections I don't see this pattern nor in our code base nor in MS one. I can add the `OrDefault` suffix if you really think we should add something.

Reply:
_Get_ doesn't suggest that it could return null, so I think adding _orDefault_ would be better as it would remind callers they need to code defensively.